File: src/backend/CareFirstClinic.API/CareFirstClinic.API/Models/Payment.cs
namespace CareFirstClinic.API.Models
{
    public enum PaymentStatus
    {
        Pending,
        Completed,
        Failed,
        Refunded
    }

    public enum PaymentMethod
    {
        Cash,
        CreditCard,
        VNPay,
        BankTransfer
    }

    public enum PaymentType
    {
        ConsultationFee,
        MedicineFee
    }

    public class Payment
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid PatientId { get; set; }
        public Guid AppointmentId { get; set; }

        public string OrderId { get; set; } = string.Empty;
        public decimal Amount { get; set; }

        public PaymentType Type { get; set; }   
        public PaymentMethod Method { get; set; } = PaymentMethod.Cash;
        public PaymentStatus Status { get; set; } = PaymentStatus.Pending;

        public string? TransactionId { get; set; }
        public string? BankCode { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? PaidAt { get; set; }

        public string? Notes { get; set; }

        public Patient? Patient { get; set; }
        public Appointment? Appointment { get; set; }
    }
}

File: src/frontend/src/types/payment.ts
export interface Payment {
  id: string;
  patientId: string;
  patientName: string;
  appointmentId: string;
  orderId: string;
  amount: number;
  type: 'ConsultationFee' | 'MedicineFee';
  method: 'Cash' | 'CreditCard' | 'VNPay' | 'BankTransfer';
  status: 'Pending' | 'Completed' | 'Failed' | 'Refunded';
  transactionId?: string;
  bankCode?: string;
  notes?: string;
  createdAt: string;
  paidAt?: string;
}

export interface CreatePaymentDTO {
  appointmentId: string;
  amount: number;
  type: 'ConsultationFee' | 'MedicineFee';
  method: 'Cash' | 'CreditCard' | 'VNPay' | 'BankTransfer';
  notes?: string;
  patientId: string;
}

export interface VNPayPaymentRequest {
  appointmentId: string;
}

export interface VNPayPaymentResponse {
  success: boolean;
  message: string;
  data?: {
    paymentUrl: string;
    orderId: string;
    paymentId: string;
    amount: number;
    expiresInMinutes: number;
  };
}

export interface VNPayReturnResult {
  isSuccess: boolean;
  orderId: string;
  transactionId: string;
  amount: number;
  responseCode: string;
  transactionStatus: string;
  bankCode: string;
  payDate: string;
  message: string;
}

export interface PaymentStatusResponse {
  orderId: string;
  paymentId: string;
  status: string;
  amount: number;
  transactionId?: string;
  bankCode?: string;
  paidAt?: string;
  createdAt: string;
}

export interface PaymentHistoryResponse {
  id: string;
  orderId: string;
  amount: number;
  status: string;
  method: string;
  transactionId?: string;
  serviceName?: string;
  createdAt: string;
  paidAt?: string;
}

export interface PaymentQueryParams {
  appointmentId?: string;
  patientId?: string;
  status?: 'Pending' | 'Completed' | 'Failed' | 'Refunded';
  method?: 'Cash' | 'CreditCard' | 'VNPay' | 'BankTransfer';
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortDir?: string;
  fromDate?: string;
  toDate?: string;
}